src/main/bugfree/Bugfree.php

```diff
@@ -45,7 +45,7 @@ public function __construct(Resolver $resolver, Config $config)
     public function parse($name, $source)
     {
         $result = new Result($name, $this->config);
-        $parser = new Parser(new Lexer());
+        $parser = (new ParserFactory())->create(ParserFactory::ONLY_PHP7, new Lexer());
```

Comment:
what about:
```
PHP_MAJOR_VERSION >= 7 ? ParserFactory::ONLY_PHP7 : ParserFactory::PREFER_PHP7
```

That way if your running on 7 you get strict checking, but if you have a legacy codebase it can keep doing 5 gracefully.